js/lcx.js

```diff
@@ -0,0 +1,579 @@
+            'type': type,
+            'side': side,
+            'status': status,
+            'price': price,
+            'amount': amount,
+            'filled': filled,
+            'remaining': remaining,
+            'average': average,
+            'cost': cost,
+            'fee': undefined,
+            'trades': undefined,
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'Pair': market['symbol'],
+            'Amount': parseFloat (this.amountToPrecision (symbol, amount)),
+            'Price': parseFloat (this.priceToPrecision (symbol, price)),
+            'OrderType': type.toUpperCase (),
+            'Side': side.toUpperCase (),
+        };
+        const response = await this.privatePostCreate (this.extend (request, params));
+        const data = this.safeValue (response, 'data');
+        return this.parseOrder (data);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        const request = {
+            'OrderId': id,
+        };
+        const response = await this.privatePostCancel (this.extend (request, params));
+        const data = this.safeValue (response, 'data');
+        return this.parseOrder (data);
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api] + '/';
+        const query = this.omit (params, this.extractParams (path));
+        if (api === 'public') {
+            if (method === 'POST') {
+                url += this.implodeParams (path, params);
+                body = this.json (query);
+            } else {
+                url += this.implodeParams (path, params);
+                if (Object.keys (query).length) {
+                    url += '?' + this.urlencode (query);
+                }
+            }
+        } else if (api === 'private') {
+            path = 'api' + '/' + path;
+            const now = this.nonce ();
+            this.checkRequiredCredentials ();
+            let payload = method + '/' + path;
+            if (method !== 'GET') {
+                payload = method + '/' + path + this.json (query);
+            }
+            const signature = this.hmac (payload, this.secret, 'sha256', 'base64');
+            headers = {
+                'x-access-key': this.apiKey,
+                'x-access-sign': signature,
+                'x-access-timestamp': now,
+            };
+            url += this.implodeParams (path, params);
+            if (method === 'GET') {
+                if (Object.keys (query).length) {
+                    url += '?' + this.urlencode (query);
+                }
+            } else if (Object.keys (query).length) {
+                body = this.json (query);
+            }
```

Comment:
This branching is not very clear. How does checking the `HTTP method` and testing `Object.keys (query).length` relate in this branching?